common/src/Microsoft.Azure.IIoT.Core/src/Utils/Logging.cs

```diff
@@ -0,0 +1,26 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.Utils
+{
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// Helper class for logging
+    /// </summary>
+    public static class Logging {
+        /// <summary>
+        /// Extracts server:port from a uri for simpler logging.
+        /// </summary>
+        /// <param name="endpointUrl"></param>
+        /// <returns></returns>
+        public static string ExtractServerPort(string endpointUrl) {
+            string pattern = @":\/\/([^\/_]+)";
+            var match = Regex.Match(endpointUrl, pattern);
```

Comment:
Why not just "new Url(endpointUrl).Port"?   Saves you all the tests, etc.?

Reply:
I renamed the file to LoggingHelper.cs and the method to ExtractHost. The name of the method was misleading as it doesn't extract the port only. Some input types such as "http://example.com:8080_hash" would not work with the Uri class.

Reply:
I think we should have this very "specialized" code somewhere closer to its actual use - and I assume it is only used once, correct?   My preference:  Create real url in publisher as identifier, and then extract what is necessary from it using the existing parsers.   E.g. _ could be just a / and we should be able to use Url(…).

Reply:
Here it's only used once, but it will be used more once Cristian is done refactoring the session management. I'm waiting for him to avoid merging overhead. This helper also gives us a way to change the formatting centrally in the future.

Reply:
Then let's move it into the same package as where it is used as it feels very "publisher-ish".

Reply:
Done